test/test_modules_taken_before_others.py

```diff
@@ -0,0 +1,120 @@
+                           'and not empty\')" oninput="setCustomValidity(\'\')" required>'
+    CONTENT_FORM_SELECT = '<select style="display:inline-block;" name="aysem" required>'
+    CONTENT_TABLE = '<table class="table display dataTable table-bordered table-hover'+\
+                    ' text-center" id="modules-taken-prior-table">'
+    CONTENT_BUTTON = '<input class="btn btn-primary" style="margin-top: '+\
+                     '10px;" type="submit" value="Go">'
+
+    FORM_NAME = 'module-filter-form'
+    REDIRECT_PAGE = '<h3 style="margin-bottom: 20px;">Students who took <b><a '+\
+                    'href="/viewModule?code=CS1010"'+\
+                    ' target="_blank" data-toggle="tooltip" title="View general info for'+\
+                    ' CO1020>CS1020</a></b> prior to taking <b><a '+\
+                    'href="/viewModule?code=$moduleB" '+\
+                    'target="_blank" data-toggle="tooltip" title="View general '+\
+                    'info for CS1020">CS1020</a></b> in <u>AY 16/17 Sem 2</u></h3>'
+
+    def __init__(self):
+        self.middleware = None
+        self.test_app = None
+
+    def  setUp(self):
+        '''
+            Sets up the 'app.py' fixture
+        '''
+        self.middleware = []
+        self.test_app = TestApp(APP.wsgifunc(*self.middleware))
+        session.set_up(self.test_app)
+
+    def tearDown(self):
+        '''
+            Tears down 'app.py' fixture and logs out
+        '''
+        session.tear_down(self.test_app)
+
+    def test_modules_taken_before_others_response(self):
+        '''
+            Tests that we can access the page
+        '''
+        root = self.test_app.get(self.URL_NORMAL)
+        assert_equal(root.status, 200)
+
+    def test_modules_taken_before_others_contents(self):
+        '''
+            Test that the contents on the page are as expected.
+        '''
+        root = self.test_app.get(self.URL_NORMAL)
+        root.mustcontain(self.CONTENT_TITLE)
+        root.mustcontain(self.CONTENT_DESCRIPTION)
+        root.mustcontain(self.CONTENT_FORM)
+        root.mustcontain(self.CONTENT_FORM_TITLE)
+        root.mustcontain(self.CONTENT_FORM_MODULEA)
+        root.mustcontain(self.CONTENT_FORM_MODULEB)
+        root.mustcontain(self.CONTENT_FORM_SELECT)
+        root.mustcontain(self.CONTENT_TABLE)
+        root.mustcontain(self.CONTENT_BUTTON)
+
+    def test_modules_taken_before_others_submit(self):
+        '''
+            tests taht the submit button will cause a redirect
+        '''
+        root = self.test_app.get(self.URL_NORMAL)
+
+        form = root.forms__get()[self.FORM_NAME]
+        form.__setitem__("moduleA", "CS1010")
+        form.__setitem__("moduleB", "CS2010")
+        form.__setitem__("aysem", "AY 16/17 Sem 2")
+
+        response = form.submit()
+        #check response is 200 OK
+        assert_equal(response.status, 200)
+        response.mustcontain()
```

Comment:
Why is there nothing passed into the mustcontain(), then what is this line checking for ?